crates/execution/src/matching_engine/engine.rs

```diff
@@ -1106,45 +1106,129 @@ impl OrderMatchingEngine {
+                        return true;
+                    }
+                    return false;
+                }
+
+                let ap = inner.activation_price.unwrap();
+                let hit = match inner.order_side() {
+                    OrderSide::Buy => ask.is_some_and(|a| a <= ap),
+                    OrderSide::Sell => bid.is_some_and(|b| b >= ap),
+                    _ => false,
+                };
+                if hit {
+                    inner.set_activated();
+                    if let Err(e) = self.cache.borrow_mut().update_order(order) {
+                        log::error!("Failed to update order: {e}");
+                    }
+                }
+                hit
+            }
+            OrderAny::TrailingStopLimit(inner) => {
+                if inner.is_activated {
+                    return true;
+                }
+
+                if inner.activation_price.is_none() {
+                    let px = match inner.order_side() {
+                        OrderSide::Buy => ask,
+                        OrderSide::Sell => bid,
+                        _ => None,
+                    };
+                    if let Some(p) = px {
+                        inner.activation_price = Some(p);
+                        inner.set_activated();
+                        if let Err(e) = self.cache.borrow_mut().update_order(order) {
+                            log::error!("Failed to update order: {e}");
+                        }
+                        return true;
+                    }
+                    return false;
+                }
+
+                let ap = inner.activation_price.unwrap();
+                let hit = match inner.order_side() {
+                    OrderSide::Buy => ask.is_some_and(|a| a <= ap),
+                    OrderSide::Sell => bid.is_some_and(|b| b >= ap),
+                    _ => false,
+                };
+                if hit {
+                    inner.set_activated();
+                    if let Err(e) = self.cache.borrow_mut().update_order(order) {
+                        log::error!("Failed to update order: {e}");
+                    }
+                }
+                hit
+            }
+            _ => true,
+        }
+    }
+
     fn iterate_orders(&mut self, timestamp_ns: UnixNanos, orders: &[PassiveOrderAny]) {
         for order in orders {
             if order.is_closed() {
                 continue;
             }
 
-            // Check expiration
-            if self.config.support_gtd_orders
-                && let Some(expire_time) = order.expire_time()
-                && timestamp_ns >= expire_time
-            {
-                // SAFTEY: We know this order is in the core
-                self.core.delete_order(order).unwrap();
-                self.cached_filled_qty.remove(&order.client_order_id());
-                self.expire_order(order);
+            if self.config.support_gtd_orders {
+                if let Some(expire_time) = order.expire_time() {
```

Comment:
The new clippy lint will probably denest this condition to `if self.config.support_gtd_orders && if let Some(expire_time) = order.expire_time()`.